### PR TITLE
Feature/chartjs annotations

### PR DESCRIPTION
**Pull request by nbalasovs**
This PR contains some Line Chart updates. Mostly focused on fetching (fake) API data from source and updating chart. Glucose-level annotations were added. Annotations can be modified in the future to match app design.
Screenshot 2021-05-14 at 11 47 12

### CODE REVIEW

File: diabetes-dashboard/src/components/LineChart.vue [hunk 1]
```diff
@@ -106,58 +150,66 @@ export default {
...
+                        range.start = now.subtract(3, 'hours');
+                        break;
+                    case '1h':
+                        range.start = now.subtract(1, 'hours');
+                        break;
+                    case '5m':
+                        range.start = now.subtract(5, 'minutes');
+                        break;
+                    default:
+                        return false;
```
Comment (JfmGijsbers) [inline diabetes-dashboard/src/components/LineChart.vue:179]: I personally don't quite like this. Maybe consider passing as "flag" { hours: x, minutes: y } and then always just subtract flag. Less code == more clear

File: diabetes-dashboard/src/components/LineChart.vue [hunk 2]
```diff
@@ -106,58 +150,66 @@ export default {
...
-                end: null
-            };
+            if (range.start && range.end) {
+                this.$emit('filtered', range);
+                this.title = this.updateTitle(window.lineChart);
+                return true;
+            }
 
-            return true;
+            return false;
```
Comment (JfmGijsbers) [inline diabetes-dashboard/src/components/LineChart.vue:193]: Are these return values actually used anywhere? If not, just remove them (if you want to exit from a function somewhere, just use "return" without a value after it

File: diabetes-dashboard/src/components/LineChart.vue [hunk 3]
```diff
@@ -18,44 +18,58 @@
...
-            chart: null,
             range: {
                 start: null,
                 end: null,
             },
             show: false,
             options: {
                 type: 'line',
-                data: { labels: this.labels, datasets: this.datasets },
+                data: this.data,
```
Comment (JfmGijsbers) [inline diabetes-dashboard/src/components/LineChart.vue:72]: The "data" value is already defined in the props, so this is a self-assign and should be removed

File: diabetes-dashboard/src/views/Dashboard.vue [hunk 4]
```diff
@@ -80,6 +80,15 @@ export default {
     LineChart
   },
   methods: {
+      // Test request that simulates receiving updated chart data, proper
+      // documentation will be required
+      updateData(value) {
+          console.log(value);
```
Comment (JfmGijsbers) [inline diabetes-dashboard/src/views/Dashboard.vue:86]: console.logs should never be in production code

File: diabetes-dashboard/src/components/LineChart.vue (hunk 3 above)
Comment (nbalasovs) [reply diabetes-dashboard/src/components/LineChart.vue:72]: I am not sure it is self-assigned because I see no chart if I remove this line

**Review by JfmGijsbers — APPROVED** [review]

**Pull request merged**